bin/iam_utils.py

```diff
@@ -42,527 +38,606 @@
-            self.roles = json.load(f)
-
-    def extract_groups_from_iam_details(self):
-        group_temp_list = []
-        for group in self.iam_details["GroupDetailList"]:
-            if self.filter("GroupName", self.group_keyword_filters, self.group_whole_filters, group):
-                print("filtering: " + group["GroupName"])
-                continue
-
-            new_group = {'GroupName': group['GroupName'],
-                        'Path': group['Path']}
-
-            managed_pol_list = []
-            for mp_pol in group["AttachedManagedPolicies"]:
-                managed_pol_list.append(mp_pol["PolicyArn"])
-            new_group["AttachedManagedPolicies"] = managed_pol_list
-
-            inst_pol_group_list = []
-            for inst_pol in group["GroupPolicyList"]:
-                doc = {}
-                doc["PolicyDocument"] = inst_pol["PolicyDocument"]
-                doc["PolicyName"] = inst_pol["PolicyName"]
-                inst_pol_group_list.append(inst_pol)
-            new_group["GroupPolicyList"] = inst_pol_group_list
-
-            group_temp_list.append(new_group)
-        self.groups = self.to_prod_account(group_temp_list)
-
-    def create_full_group(self, group):
-        self.iw.create_group(group["GroupName"], group["Path"])
-        for inline_pol in group["GroupPolicyList"]:
-            self.iw.put_group_policy(group["GroupName"], inline_pol["PolicyName"], inline_pol["PolicyDocument"])
-        for mngd_pol_arn in group["AttachedManagedPolicies"]:
-            self.iw.attach_group_policy(group["GroupName"], mngd_pol_arn)
-
-    def adjust_groups_in_aws(self, use_assume_role=False):
-        '''
-        Adjusts the AWS groups to match the current in memory groups. It creates the groups if
-        they do not exist in AWS.  Also adjusts attached managed policies and inline policies associated with the group
-         It will not delete any groups.
+            raise ValueError("resource_type '{}' is not supported".format(resource_type))
+
+        resources = []
+
+        kwargs = { 'MaxItems': 1000, 'Filter': filter }
+        resp = {'IsTruncated': True}
+        while resp['IsTruncated']:
+            resp = self.client.get_account_authorization_details(**kwargs)
+            kwargs['Marker'] = resp.get('Marker')
+
+            resources.extend([self.extract(resource_type, item)
+                              for item in resp[list_key]
+                              if item[name_key] in names])
+
+        return resources
+
+    def extract(self, resource_type, resource):
+        """Convert the IAM object into the internal representation used
+
         Args:
-            use_assume_role: set to True if using developer account credentials and plan to assume production credentials
+            resource_type (str): One of - groups, roles, policies
+            resource (object): IAM object
 
         Returns:
+            object: Converted IAM object
+        """
+        key = {
```

Comment:
Doesn't look like this is used, anymore.

Reply:
I think you are correct, removed